ocean_lib/data_provider/base.py

```diff
@@ -0,0 +1,310 @@
+    @enforce_types
+    def build_download_endpoint(provider_uri: str) -> Tuple[str, str]:
+        return DataServiceProviderBase.build_endpoint("download", provider_uri)
+
+    @staticmethod
+    @enforce_types
+    def build_compute_endpoint(provider_uri: str) -> Tuple[str, str]:
+        return DataServiceProviderBase.build_endpoint("computeStatus", provider_uri)
+
+    @staticmethod
+    @enforce_types
+    def build_compute_result_file_endpoint(provider_uri: str) -> Tuple[str, str]:
+        return DataServiceProviderBase.build_endpoint("computeResult", provider_uri)
+
+    @staticmethod
+    @enforce_types
+    def build_fileinfo(provider_uri: str) -> Tuple[str, str]:
+        return DataServiceProviderBase.build_endpoint("fileinfo", provider_uri)
+
+    @staticmethod
+    @enforce_types
+    def build_c2d_environments_endpoint(provider_uri: str) -> Tuple[str, str]:
+        return DataServiceProviderBase.build_endpoint(
+            "computeEnvironments", provider_uri
+        )
+
+    @staticmethod
+    @enforce_types
+    def write_file(
+        response: Response,
+        destination_folder: Union[str, bytes, os.PathLike],
+        file_name: str,
+    ) -> None:
+        """
+        Write the response content in a file in the destination folder.
+        :param response: Response
+        :param destination_folder: Destination folder, string
+        :param file_name: File name, string
+        :return: None
+        """
+        if response.status_code == 200:
+            with open(os.path.join(destination_folder, file_name), "wb") as f:
+                for chunk in response.iter_content(chunk_size=4096):
+                    f.write(chunk)
+            logger.info(f"Saved downloaded file in {f.name}")
+        else:
+            logger.warning(f"consume failed: {response.reason}")
+
+    @staticmethod
+    @enforce_types
+    def _get_file_name(response: Response) -> Optional[str]:
+        try:
+            return re.match(
+                r"attachment;filename=(.+)", response.headers.get("content-disposition")
+            )[1]
+        except Exception as e:
+            logger.warning(f"It was not possible to get the file name. {e}")
+            return None
+
+    @staticmethod
+    @enforce_types
+    def _http_method(method: str, *args, **kwargs) -> Optional[Union[Mock, Response]]:
+        try:
+            return getattr(DataServiceProviderBase._http_client, method)(
+                *args, **kwargs
+            )
+        except Exception:
+            logger.error(
+                f"Error invoking http method {method}: args={str(args)}, kwargs={str(kwargs)}"
+            )
+            raise
+
+    @staticmethod
+    @enforce_types
+    def check_response(
+        response: Any,
+        endpoint_name: str,
+        endpoint: str,
+        success_codes: Optional[List] = None,
+        exception_type=DataProviderException,
+    ):
+        if not response or not hasattr(response, "status_code"):
```

Comment:
In `check_response` we can check if the response is None.

Reply:
done, I forgot to paste something from the previous PR

Reply:
changed